spinalcordtoolbox/scripts/sct_compute_snr.py

```diff
@@ -185,6 +198,13 @@ def main(argv=None):
     # Display result
     if fname_mask:
         printv('\nSNR_' + method + ' = ' + str(snr_roi) + '\n', type='info')
+    
+    #Added function for text file
+    if file_name is not None:
+        f = open(file_name,"w")
+        f.write(str(snr_roi))
+        f.close
+        printv('\nFile saved to '+ file_name)
 
 
```

Comment:
One other suggestion: Could you please update the existing test? I've linked to the existing test file below:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/24b6ca2511e9a4163da1fdb7565f9a39bbf1d267/testing/cli/test_cli_sct_compute_snr.py#L11-L15

We can add the `-o` argument to this test, then check to make sure the computed SNR matches an expected value. Here is a rough idea of what the updated test could look like:

```python
 def test_sct_compute_snr_against_groundtruth(): 
     """Run the CLI script and check SNR against a ground truth value.""" 
     fname_out = "computed_snr.txt"
     sct_compute_snr.main(argv=['-i', 'dmri/dwi.nii.gz', '-m', 'dmri/dmri_T0001.nii.gz', '-method', 'diff', 
                                '-vol', '0,5', '-o', fname_out])
    with open(fname_out, "r") as f:
         snr = f.read()
    assert snr == pytest.approx(1.234) # Replace with whatever the SNR is supposed to be
``` 

We also have the [Testing](https://github.com/spinalcordtoolbox/spinalcordtoolbox/wiki/Testing) page in our developer wiki which provides some tips for writing new tests. (But, this is a simple test, so you probably won't need to do anything too complicated.)